interpreter/exec/ixx.ml

```diff
@@ -39,6 +39,9 @@ sig
   val to_int64: t -> int64
   val to_string : t -> string
   val to_hex_string : t -> string
+  (* If bit (bitwidth - 1) is set, sx will sign-extend t to maintain the
+   * invariant that small ints are stored sign-extended inside a wider int. *)
+  val sx : t -> t
```

Comment:
Is this actually needed as a parameter? I think you can implement this generically inside the module:
```
sx i = of_i64 ((to_i64 i << (64 - bitwidth)) >> (64 - bitwidth))
```

Reply:
You're right, not needed, but i can pass Fun.id in the i32 and i64 case which makes it a tiny bit more performant?